core/src/main/scala/org/http4s/util/package.scala

```diff
@@ -24,7 +26,8 @@ package object util {
           else Pull.output1(outputString).as(None)
         case Some((chunk, stream)) =>
           if (chunk.nonEmpty) {
-            val bytes = chunk.toArray
+            val chunkWithoutBom = skipByteOrderMark(chunk)
```

Comment:
This is probably fast enough as is, but do we only need to check this on the first chunk?  And is it incorrect if we drop it in the middle of a stream?

Reply:
Yes, checking only the first chunk is enough.

According to [wikipedia](https://en.wikipedia.org/wiki/Byte_order_mark#UTF-8):
> its only use in UTF-8 is to signal at the start that the text stream is encoded in UTF-8, or that it was converted to UTF-8 from a stream that contained an optional BOM.

So in both implementations (here and in fs2) if UTF-8 BOM is not at the very beginning of the stream - it will not be dropped and will remain in the result.

So we could optimize it possibly introducing some var to mark the first chunk... But hopefully that is not necessary.